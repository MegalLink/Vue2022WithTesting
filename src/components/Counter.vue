<template>
  <div class="container-counter">
    <h2>{{ getTitle }}</h2>
    <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
    <button class="btn" @click="increase">+1</button>
    <button class="btn" @click="decrease">-1</button>
  </div>
</template>
<script>
export default {
  // name: 'OtherName' in developer tools apears with this name
  props: {
    title: String,
    start: {
      type: Number,
      required: true,
      default: 10,
      validator(value) {
        return value > 5;
      },
    },
  },
  data() {
    return {
      counter: this.start,
    };
  },
  methods: {
    increase() {
      this.counter++;
    },
    decrease() {
      this.counter--;
    },
  },
  computed: {
    getSquareCounter() {
      return this.counter * this.counter;
    },
    getTitle() {
      return this.title || "Counter";
    },
  },
};
</script>
<style scoped>
.btn {
  margin: 0 10px;
  background-color: rgb(231, 131, 0);
  color: white;
  border-radius: 5px;
  border-color: transparent;
}

.btn:hover {
  background-color: rgb(238, 175, 93);
  transition: 0.3s ease-in-out;
}

.container-counter {
  position: relative;
  background: black;
  padding: 2rem;
  z-index: 100;
  margin-bottom: 20px;
}
</style>
