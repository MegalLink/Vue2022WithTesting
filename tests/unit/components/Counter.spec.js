import Counter from "@/components/Counter";
import { shallowMount } from "@vue/test-utils";
describe("Counter components", () => {
  let wrapper;
  let buttons;
  beforeEach(() => {
    wrapper = shallowMount(Counter);
    buttons = wrapper.findAll("button");
  });

  it("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render deafult title 'Counter'", () => {
    let h2 = wrapper.find('[test-id="title"]');
    expect(h2.text()).toBe("Counter");
  });

  it("should render have default counter value '10'", () => {
    let span = wrapper.find("span");
    expect(span.text()).toBe("10");
  });

  it("should increase counter by one", async () => {
    await buttons[0].trigger("click");
    let counterText = wrapper.find("span").text();
    expect(counterText).toBe("11");
  });

  it("should decrease counter by one", async () => {
    await buttons[1].trigger("click");
    let counterText = wrapper.find("span").text();
    expect(counterText).toBe("9");
  });
});
