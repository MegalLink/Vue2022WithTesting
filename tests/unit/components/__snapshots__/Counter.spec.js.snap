// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter components should match snapshot 1`] = `
<div class="container-counter">
  <h2 test-id="title">Counter</h2>
  <p><span>10</span> <sup>2</sup> = 100</p><button class="btn">+1</button><button class="btn">-1</button>
</div>
`;
